### ISSUE TITLE
[Bug]: metamask bug at opbnb network

### ISSUE THREAD

**Issue by differs** [MetaMask/metamask-extension#22990]
### Describe the bug

while converting number to string, invalid number value '1.1e-9', should be a number matching (^-?[0-9.]+).

### Expected behavior

on OPBNB network the gas price is 9 wei. metamask can't process it in right way.

### Screenshots/Recordings

![Screenshot from 2024-02-16 15-19-39](https://github.com/MetaMask/metamask-extension/assets/16659721/dfd5608d-8d53-464c-a92b-0f78850e3575)

### Steps to reproduce

1. select opbnb network
2. send bnb to other address
3. modify the gas price to 9 wei
4. the errors will replay and you can't send assert forever

### Error messages or log output

```shell
while converting number to string, invalid number value '1.1e-9', should be a number matching (^-?[0-9.]+).
```

### Version

11.9.4

### Build type

None

### Browser

Chrome, Firefox, Microsoft Edge, Brave

### Operating system

Windows, MacOS, Linux

### Additional context

while converting number to string, invalid number value '1.1e-9', should be a number matching (^-?[0-9.]+).

**Comment by anaamolnar** [comment]
Related to https://github.com/MetaMask/metamask-extension/issues/22605

**Comment by vinistevam** [comment]
Thank you so much for reporting this issue @differs .
A [similar issue](https://github.com/MetaMask/metamask-extension/issues/22605) was fixed recently and it will be available in the `release-11.12.0`. I've tested the fix and it is working for `opbnb` network.

https://github.com/MetaMask/metamask-extension/assets/45455812/d0f784eb-b0dc-42cd-bb17-80dab3ca154b

Let me know if you need anything else.

**Comment by dbrans** [comment]
Closing the issue as fixed. Don't hesitate to reopen if there's anything else!
